app/ts/PreferencesTab/Preferences/PartialValueFunctions/AdvancedPartialValueFunctions/AdvancedPvfButtons/AdvancedPvfButtons.tsx

```diff
@@ -0,0 +1,65 @@
+import {Button, ButtonGroup, Grid} from '@material-ui/core';
+import {IPieceWiseLinearPvf} from '@shared/interface/Pvfs/IPieceWiseLinearPvf';
+import {PreferencesContext} from 'app/ts/PreferencesTab/PreferencesContext';
+import DisplayWarnings from 'app/ts/util/DisplayWarnings';
+import {SubproblemContext} from 'app/ts/Workspace/SubproblemContext/SubproblemContext';
+import _ from 'lodash';
+import React, {useContext} from 'react';
+import {AdvancedPartialValueFunctionContext} from '../AdvancedPartialValueFunctionContext/AdvancedPartialValueFunctionContext';
+
+export default function AdvancedPvfButtons(): JSX.Element {
+  const {setActiveView, setPieceWisePvf} = useContext(PreferencesContext);
+  const {configuredRanges} = useContext(SubproblemContext);
+  const {advancedPvfCriterion, direction, cutOffs, isSaveDisabled} = useContext(
+    AdvancedPartialValueFunctionContext
+  );
+
+  const configuredRange =
```

Comment:
Should we make a getConfiguredRange(criterion) for Subproblem context? This code many places, such repeat.

Reply:
makes sense